File: benchmarks/oopsla25/subjects/333.c
#include <assert.h>
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

void f333(const char* s) {
    if (strlen(s) > 0) {
        char a = s[0];
        char x = s[1];
        assert(a == 'a');
    }
    if (strlen(s) > 1) {
        char y = s[1];
        char b = s[2];
        assert(b == 'b');
    }
    if (strlen(s) > 3) {
        exit(1);
    }
}

int main(int argc, char *argv[]) {
  char my_string[10240] = {0};
  if (argc == 1) {
    read(fileno(stdin), my_string, 10240);
  } else {
    int fd = open(argv[1], O_RDONLY);
    read(fd, my_string, 10240);
    close(fd);
  }
  f333(my_string);
  return 0;
}

File: tests/oopsla25/333.py
#!/usr/bin/env python3

import sys

def f333(s: str):    
  if len(s) > 0:
    a,x = s[0:2]
    assert a == "a"
  if len(s) > 1:
    y,b = s[1:3]  
    assert b == "b"
  if len(s) > 3:
    raise Exception

if __name__ == '__main__':
  my_string = ""
  if len(sys.argv) == 1:
    my_string = sys.stdin.read()
  else:
    fd = open(sys.argv[1], 'r')
    my_string = fd.read()
    fd.close()
  
  f333(my_string)
